### ISSUE TITLE
Refactor try-catch block to control flow

### ISSUE THREAD

**Issue by Matthew-Boyd** [usnistgov/teqp#126]
Can this try-catch block be refactored into a control flow block so exceptions are not thrown by design when JSON representations are passed instead of a path?

https://github.com/usnistgov/teqp/blob/201ceb7c59780bd93df484b90c2b2e986823116f/include/teqp/ideal_eosterms.hpp#L494-L501

**Comment by ianhbell** [comment]
Is there a problem with the current code, or just a question of elegance?

**Comment by Matthew-Boyd** [comment]
Mostly the latter, but I'd like to eliminate "Exception thrown" messages from stdout so they don't obscure new occurrences.

**Comment by ianhbell** [comment]
You mean if the catch branch is hit, and the parsing fails, that exception? What is your preferred error handling here?

**Comment by ianhbell** [comment]
I'm not entirely following. What do you want the code to look like?

**Comment by Matthew-Boyd** [comment]
create path from string argument
if path points to a valid file (e.g., is_regular_file())
  try calling load_a_JSON_file()
else if path is not valid (e.g., empty())
  try json::parse()
else
  error

**Comment by ianhbell** [comment]
And what does error handling look like in your pseudocode? 

Alternatively, we could also support the prefix that I have started to implement elsewhere to specify what it is you have in the string and avoid some of the inference: https://github.com/usnistgov/teqp/blob/201ceb7c59780bd93df484b90c2b2e986823116f/src/tests/catch_test_multifluid_RPinterop.cxx#L45

**Comment by Matthew-Boyd** [comment]
Whatever integrates with what you're already doing. This overall change is
to not have an exception as a matter of course.

**Comment by ianhbell** [comment]
Ok I remember now why I had to do things the way I did. ``std::filesystem::is_regular_file`` throws if the filename is too long (as blocks of JSON-encoded data tend to be), which scuppers the nice control flow you desire. Hence one must try and catch. Figuring out the longest allowed path is not trivial either: https://stackoverflow.com/questions/65174086/is-there-any-difference-between-filename-max-and-path-max-in-c
